org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteDatasetInfo.java

```diff
@@ -0,0 +1,77 @@
+package org.eclipse.dawnsci.remotedataset.client;
+
+import java.util.List;
+
+import org.eclipse.january.dataset.ShapeUtils;
+
+public class RemoteDatasetInfo {
+	
+	private String name;
+	private int[] shape;
+	private int[] maxShape;
+	private int dtype;
+	private int isize;
+	private int[] chunking;
+
+	public RemoteDatasetInfo(List<String> info) {
+
+		this.name  = info.get(0);
+		this.shape = toIntArray(info.get(1));
+		if (this.shape.length == 1 && shape[0] == -1) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "shape" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)